spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.spec.js

```diff
@@ -644,5 +718,70 @@ describe('GOVUK.analyticsGa4.linkTracker', function () {
+      '<a href="https://example.com/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui/https://example.com/test">check regex replace only replaces first domain match</a>' +
+      '<a href="https://assets.publishing.service.gov.uk/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui">check regex replace works with a long subdomain </a>' +
+      '<a href="//assets.publishing.service.gov.uk/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui">check regex replace works with protocal relative domain </a>' +
+      '</div>'
+
+      var expectedLinkPathParts = [
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-',
+          2: 'say-it-loud-enough-youll-always-sound-precocious-supercalifragilisticexpialidocious-supercalifragili'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-',
+          2: 'say-it-loud-enough-youll-always-sound-precocious-supercalifragilisticexpialidocious-supercalifragili',
+          3: 'sticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-say-it-loud-enoug',
+          4: 'h-youll-always-sound-precocious-supercalifragilisticexpialidocious-supercalifragilisticexpialidociou',
+          5: 's-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-say-it-loud-enough-youll-always-so'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-',
+          2: 'say-it-loud-enough-youll-always-sound-precocious-supercalifragilisticexpialidocious-supercalifragili',
+          3: 'sticexpialidocious-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-say-it-loud-enoug',
+          4: 'h-youll-always-sound-precocious-supercalifragilisticexpialidocious-supercalifragilisticexpialidociou',
+          5: 's-even-though-the-sound-of-it-is-something-quite-atrocious-if-you-say-it-loud-enough-youll-always-so'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui/https://example.com',
+          2: '/test'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui'
+        },
+        {
+          1: '/supercalifragilisticexpialidocious-even-though-the-sound-of-it-is-something-qui'
+        }
+      ]
+
+      var linksToTest = document.querySelectorAll('.long-links a')
+      for (var i = 0; i < linksToTest.length; i++) {
+        defaultLinkPathParts = {
```

Comment:
This object is being defined multiple times throughout this file - could this bit be moved to the top level `beforeEach`?

Reply:
@andysellick It can't because it needs to be reset before each link click, instead of before each test

Reply:
Ah yes, sorry, missed that. However if you leave this one in place (I see now that it resets each time through the loop) you can remove all the other instances in the rest of the file (e.g. line 511, 579, etc.) and put those all in a new `beforeEach` at the start of the file.